lib/uo_cli.rb

```diff
@@ -677,6 +693,57 @@ def self.create_project_folder(dir_name, empty_folder = false, overwrite_project
       end
     end
 
+    # Update an existing URBANopt Project
+    # params\
+    # +existing_project_folder+:: _string_ Name of existing project folder
+    # +new_project_directory+:: _string_ Location of updated URBANopt project
+    #
+    # Includes weather for example location, a base workflow file, and mapper files to show a baseline and a high-efficiency option.
+    def self.update_project(existing_project_folder, new_project_directory)
+      path = File.expand_path(existing_project_folder)
+      new_path = Pathname.new(new_project_directory)
+
+      if Dir.exist?(new_path)
+        FileUtils.rm_rf(new_path)
+      end
+
+      FileUtils.copy_entry(path, new_path)
+
+      $LOAD_PATH.each do |path_item|
+        if path_item.to_s.end_with?('example_files')
+          # Replace mappers
+          mappers = File.join(path_item, 'mappers')
+          Pathname.new(mappers).children.each { |mapper| FileUtils.cp_r(mapper, File.join(new_path, 'mappers'), remove_destination: true) }
+
+          # Replace visualization files
+          visualization = File.join(path_item, 'visualization')
+          Pathname.new(visualization).children.each { |viz| FileUtils.cp_r(viz, File.join(new_path, 'visualization'), remove_destination: true) }
+
+          # Replace residential files
+          if Dir.exist?(File.join(path, 'residential'))
+            Pathname.new(File.join(path_item, 'residential')).children.each { |res| FileUtils.cp_r(res, File.join(new_path,'mappers', 'residential'), remove_destination: true) }
+          end
+
+          # Replace Reopt assumption files
+          if Dir.exist?(File.join(path, 'reopt'))
+            Pathname.new(File.join(path_item, 'reopt')).children.each { |reopt_file| FileUtils.cp_r(reopt_file, File.join(new_path, 'reopt'), remove_destination: true) }
+          end
+
+          # Replace OpenDSS files
+          if Dir.exist?(File.join(path, 'opendss'))
+            Pathname.new(File.join(path_item, 'opendss')).children.each { |opendss_file| FileUtils.cp_r(opendss_file, File.join(new_path, 'opendss'), remove_destination: true) }
+          end
+
+          # Replace DISCO files
+          if Dir.exist?(File.join(path, 'disco'))
+            Pathname.new(File.join(path_item, 'disco')).children.each { |disco_file| FileUtils.cp_r(disco_file, File.join(new_path, 'disco'), remove_destination: true) }
+          end
+
+          # TODO: Replace geojson files?
```

Comment:
@kflemin any thoughts on how we should copy the feature file? These would vary by the project we create, should we determine the project time and then copy the relevant one?

Reply:
yes I think that would work: check that each "example" project json file exists, if it does, replace with the newer copy?